### PR TITLE
kafka(ticdc): kafka sink support compression after encoding event into message

### LINKED ISSUE #9596: kafka sink support compress the message value

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #9596 

### What is changed and how it works?

* After encoding the event into message, compress the value before compare against the message size limit.
* Update the monitoring to track the compression ratio, also record the number of records carried by the request.

image

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
Kafka sink support compress the message value before compare against the message size limit to reduce large message count.
```

### CODE REVIEW

File: pkg/compression/compress.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+func Supported(cc string) bool {
+	switch cc {
+	case None, Snappy, LZ4:
+		return true
+	}
+	return false
+}
+
+// Encode the given data by the given compression codec.
+func Encode(cc string, data []byte) ([]byte, error) {
```
Comment (nongfushanquan) [inline pkg/compression/compress.go:45]: Why not use decompress and compress  to name the methods directly?

Comment (3AceShowHand) [reply pkg/compression/compress.go:45]: It's conventional to use `Encode` and `Decode`.

**Review by nongfushanquan — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**